platform/darwin/src/MGLGeoJSONSource.h

```diff
@@ -99,6 +99,19 @@ extern const MGLGeoJSONSourceOption MGLGeoJSONSourceOptionSimplificationToleranc
  specification</a>.
  
  @param identifier A string that uniquely identifies the source.
+ @param feature A feature that conform to the `MGLFeature` protocol.
+ @param options An `NSDictionary` of options for this source.
+ @return An initialized GeoJSON source.
+ */
+- (instancetype)initWithIdentifier:(NSString *)identifier feature:(nullable id<MGLFeature>)feature options:(nullable NS_DICTIONARY_OF(MGLGeoJSONSourceOption, id) *)options NS_DESIGNATED_INITIALIZER;
```

Comment:
Since `mbgl::style::GeoJSONSource::setGeoJSON()` takes an `mbgl::GeoJSON` (a variant of `geometry`, `feature`, or `feature_collection`), it would be better for this initializer to take any MGLShape. All MGLShape subclasses in turn have subclasses that conform to MGLFeature, so we wouldn’t be losing any expressiveness that way.